### ISSUE TITLE
Rule Request: Unnecessary closure used in stored property default value with single expression

### ISSUE THREAD

**Issue by daltonclaybrook** [realm/SwiftLint#3486]
### New Issue Checklist

- [x] Updated SwiftLint to the latest version
- [x] I searched for [existing GitHub issues](https://github.com/realm/SwiftLint/issues)

(I feel like this one is a bit challenging to search for, so my apologies if there's already an issue or if it's already implemented)

### New rule request

Possible rule identifier: `unnecessary_property_closure` (open to suggestions)

When a stored property's default value requires some customization/setup, it's often useful to use a closure as follows:

```swift
let closeButton: UIButton = {
    let button = UIButton()
    button.setImage(UIImage(named: "close"), for: .normal)
    return button
}()
```

But when that setup is done using a single expression, the closure is unnecessary:

```swift
// Unnecessary closure - should trigger a warning
let closeButton: UIButton = {
    return UIButton() 
}()

// More concise
let closeButton = UIButton()
```

This new rule should warn when using such an unnecessary closure with a single expression. I don't feel strongly one way or another about whether this should be enabled by default. This rule could probably support autocorrect as well.
